dali/kernels/slice/slice_flip_normalize_permute_cpu.h

```diff
@@ -0,0 +1,220 @@
+    (void) mean;
+    (void) inv_stddev;
+    destination = clamp<OutputType>(element);
+  }
+};
+
+struct NormalizePolicy {
+  template <typename OutputType, typename InputType>
+  static inline void Fill(OutputType &destination, InputType element,
+                          const float *mean, const float *inv_stddev) {
+    destination = clamp<OutputType>(
+      (static_cast<float>(element) - (*mean)) * (*inv_stddev));
+  }
+};
+
+template <typename OutputType, size_t Dims>
+inline void ZeroPad(OutputType *output,
+                    const std::array<int64_t, Dims> out_strides,
+                    const std::array<int64_t, Dims> padded_out_shape,
+                    std::integral_constant<size_t, 1>) {
+  constexpr auto d = Dims - 1;
+  // zero pad
+  for (int64_t i = 0; i < padded_out_shape[d]; i++) {
+    *output = 0;
+    output += out_strides[d];
+  }
+}
+
+template <typename OutputType, size_t Dims, size_t DimsLeft>
+inline void ZeroPad(OutputType *output,
+                    const std::array<int64_t, Dims> out_strides,
+                    const std::array<int64_t, Dims> padded_out_shape,
+                    std::integral_constant<size_t, DimsLeft>) {
+  constexpr auto d = Dims - DimsLeft;
+  // zero pad
+  for (int64_t i = 0; i < padded_out_shape[d]; i++) {
+    ZeroPad(output, out_strides, padded_out_shape,
+            std::integral_constant<size_t, DimsLeft - 1>());
+    output += out_strides[d];
+  }
+}
+
+template <typename Policy, bool IsNormalizationDim, typename OutputType, typename InputType,
+          size_t Dims>
+inline void SliceFlipNormalizePermuteImpl(OutputType *output, const InputType *input,
+                                          const std::array<int64_t, Dims> in_strides,
+                                          const std::array<int64_t, Dims> out_strides,
+                                          const std::array<int64_t, Dims> out_shape,
+                                          const std::array<int64_t, Dims> padded_out_shape,
+                                          const float *mean, const float *inv_stddev,
+                                          size_t normalization_dim,
+                                          std::integral_constant<size_t, 1>) {
+  constexpr auto d = Dims - 1;
+  int64_t i = 0;
+  for (; i < out_shape[d]; i++) {
+    const size_t norm_idx = IsNormalizationDim ? i : 0;
+    Policy::Fill(*output, *input, mean + norm_idx, inv_stddev + norm_idx);
+    input += in_strides[d];
+    output += out_strides[d];
+  }
+
+  // zero pad
+  for (; i < padded_out_shape[d]; i++) {
+    *output = 0;
+    output += out_strides[d];
+  }
+}
+
+template <typename Policy, bool IsNormalizationDim, typename OutputType, typename InputType,
+          size_t Dims, size_t DimsLeft>
+inline void SliceFlipNormalizePermuteImpl(OutputType *output, const InputType *input,
+                                          const std::array<int64_t, Dims> in_strides,
```

Comment:
I wonder if removing these `const`s would hurt performance in any way - it would help readability, though. Now it looks too much like a you forgot the reference - to the point where someone might try to "fix" it...

Reply:
fixed (it doesn't hit performance)